join.sh

```diff
@@ -8,6 +8,13 @@ fi
 
 set -ex
 
+#Set working directory relative to script
+SCRIPT_DIR=$(dirname $0)
+pushd $SCRIPT_DIR
+
+#Load Variables File
+. variables.sh
```

Comment:
I think I'd like to avoid passing (effectively) 2 separate configuration files in.

What if we approach this with a whitelist of environment variables, or go by some programmatic prefix (e.g. `DOCKER_` as you have used anyway) and just "always" write an environment config file to `/etc/systemd/system/docker.service.d/promenade.conf` or similar?

The most convenient thing might be a whitelist, so that operators just have to  `export HTTP_PROXY=... HTTPS_PROXY=... NO_PROXY=...` and then launch the script.

I was trying to think about how to push this inside the docker container, but I think we should support this for pulling the promenade image itself.